SIPS/sip-22.md

```diff
@@ -0,0 +1,119 @@
+---
+sip: 22
+title: Deep Links
+status: Draft
+author: Christian Montoya (@Montoya)
+created: 2024-03-20
+---
+
+## Abstract
+
+This SIP describes a URL scheme for Snaps entry points. This scheme can be used by Snap or dapp methods to trigger the entry point and navigate the user to that location. 
+
+## Motivation
+
+While some Snap entry points are reactive, meaning they can be triggered by a method, others can only be triggered by a user navigating to the location where that entry point is displayed. In order for a Snap or dapp to programmatically navigate a user to that entry point, the Snap or dapp needs a way to identify that location. The location is identified by a URL scheme, which is called a _deep link_.
+
+## Specification
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### URL Scheme 
+
+A URL for a deep link uses the format: 
+
+`metamask://[Snap ID]/page/[page name]`
+
+Where: 
+
+- `metamask` refers to the host application 
+- `[Snap ID]` refers to the canonical ID of the installed Snap
+- `page` refers to a page inside of the application 
+- `[page name]` refers to the canonical name of an individual page 
+
+### Method for Navigating to a Deep Link
+
+In order for a Snap or dapp to navigate a user to a deep link, a method is provided: 
+`snap_open`. 
```

Comment:
`snap_*` methods are only supported by Snaps. We may want to have a `wallet_*` method instead?

Reply:
Actually this makes me think the Snap should have the method internally and expose a custom RPC method to dapps if it wants to allow dapps to do this.

Reply:
I made the method specific to Snaps but I am not actually certain whether it would be best to make it a `snap_*` or `wallet_*` method. Would like to hear your perspective based on the latest draft.

Reply:
My thoughts regarding this is if we want to expose this functionality to dapps, then we should make this a `wallet_*` method. Reason is because otherwise you end up having an extraneous request to the snap. It now has to be called for the invoke snap and then it's called for the `onHomePage` call.